### PR TITLE
Montonio member payments

### PR DESCRIPTION
**Pull request by JordanValdma**

### CODE REVIEW

File: src/main/java/ee/tuleva/onboarding/payment/PaymentData.java [hunk 1]
```diff
@@ -20,14 +20,16 @@ public class PaymentData {
...
   }
 
   public enum Bank {
     LUMINOR,
     SEB,
     SWEDBANK,
     LHV,
-    COOP
+    COOP,
+    TULUNDUSUHISTU
```
Comment (ErkoRisthein) [inline src/main/java/ee/tuleva/onboarding/payment/PaymentData.java:33]: Any other way to model this? 🤔 Tulundusühistu is not a bank.

File: src/main/java/ee/tuleva/onboarding/payment/PaymentController.java [hunk 2]
```diff
@@ -38,11 +42,32 @@ public PaymentLink getPaymentLink(
   @Operation(summary = "Redirects user to payment success")
   public RedirectView getPaymentSuccessRedirect(
       @RequestParam("payment_token") String serializedToken) {
-    Optional<Payment> payment = paymentService.processToken(serializedToken);
-    if (payment.isPresent()) {
-      return new RedirectView(frontendUrl + "/3rd-pillar-success");
+    Optional<Payment> paymentOptional = paymentService.processToken(serializedToken);
+
+    return paymentOptional
```
Comment (ErkoRisthein) [inline src/main/java/ee/tuleva/onboarding/payment/PaymentController.java:47]: We should keep this kind of business logic out of the controller.

File: src/main/java/ee/tuleva/onboarding/payment/recurring/RecurringPaymentService.java [hunk 3]
```diff
@@ -65,6 +65,8 @@ public PaymentLink getPaymentLink(PaymentData paymentData, Person person) {
               + "&MakseSagedus=3" // Monthly
               + "&MakseEsimene="
               + tenthDayOfMonth(LocalDate.now(clock));
+          case TULUNDUSUHISTU -> throw new IllegalArgumentException(
+              "Can not make a recurring payment to a given bank!");
```
Comment (ErkoRisthein) [inline src/main/java/ee/tuleva/onboarding/payment/recurring/RecurringPaymentService.java:69]: "Recurring payments to the specified bank are not supported"

File: src/main/resources/application.yml [hunk 4]
```diff
@@ -238,6 +237,10 @@ payment-provider:
       access-key: f5cd942d-5b23-4467-80ce-95cd67186c6f
       secret-key: ${PAYMENT_SECRET_COOP}
       bic: EKRDEE22
+    tulundusuhistu:
+      access-key: 02f596e6-d6b6-43a8-83ac-53d33cc1f13d
+      secret-key: ${PAYMENT_SECRET_TULUNDUSUHISTU}
```
Comment (ErkoRisthein) [inline src/main/resources/application.yml:242]: Is the environment variable set up in production?

File: src/main/resources/db/migration/V1_68__payment_type.sql [hunk 5]
```diff
@@ -0,0 +1,2 @@
+DELETE FROM payment;
```
Comment (ErkoRisthein) [inline src/main/resources/db/migration/V1_68__payment_type.sql:1]: ⛔ 🚫 🖐🛑 No!!! Data loss in production!

File: src/main/java/ee/tuleva/onboarding/user/UserService.java [hunk 6]
```diff
@@ -75,6 +78,8 @@ public User registerAsMember(Long userId) {
 
     user.setMember(newMember);
 
+    applicationEventPublisher.publishEvent(new MemberCreatedEvent(user));
```
Comment (ErkoRisthein) [inline src/main/java/ee/tuleva/onboarding/user/UserService.java:81]: Maybe we emit put this event _after_ a successful save?

File: src/main/resources/db/migration/V1_68__payment_type.sql [hunk 7]
```diff
@@ -0,0 +1,2 @@
...
+ALTER TABLE payment ADD COLUMN payment_type VARCHAR(255) NOT NULL;
```
Comment (ErkoRisthein) [inline src/main/resources/db/migration/V1_68__payment_type.sql:2]: Migrate data from somewhere instead. Or use a nullable field. Or use a default value.

File: src/main/resources/application.yml (hunk 4 above)
Comment (JordanValdma) [reply src/main/resources/application.yml:242]: yes

File: src/main/resources/db/migration/V1_68__payment_type.sql (hunk 5 above)
Comment (JordanValdma) [reply src/main/resources/db/migration/V1_68__payment_type.sql:1]: Thanks!!!

File: src/main/java/ee/tuleva/onboarding/payment/PaymentData.java (hunk 1 above)
Comment (JordanValdma) [reply src/main/java/ee/tuleva/onboarding/payment/PaymentData.java:33]: I think it makes sense now, we might want to rename the entity from Bank to PaymentChannel

File: src/main/java/ee/tuleva/onboarding/payment/PaymentController.java (hunk 2 above)
Comment (JordanValdma) [reply src/main/java/ee/tuleva/onboarding/payment/PaymentController.java:47]: It's basically just routing based on Payment

File: src/main/java/ee/tuleva/onboarding/payment/recurring/RecurringPaymentService.java (hunk 3 above)
Comment (JordanValdma) [reply src/main/java/ee/tuleva/onboarding/payment/recurring/RecurringPaymentService.java:69]: Thanks

File: src/main/java/ee/tuleva/onboarding/user/UserService.java (hunk 6 above)
Comment (JordanValdma) [reply src/main/java/ee/tuleva/onboarding/user/UserService.java:81]: Good point! Thanks

File: src/main/resources/db/migration/V1_68__payment_type.sql (hunk 7 above)
Comment (JordanValdma) [reply src/main/resources/db/migration/V1_68__payment_type.sql]: Thanks
Defaulting previous payments to SINGLE, keeping NOT NULL

File: src/main/java/ee/tuleva/onboarding/payment/PaymentController.java (hunk 2 above)
Comment (JordanValdma) [reply src/main/java/ee/tuleva/onboarding/payment/PaymentController.java:47]: TODO: User redirect URL to avoid if statements in this controller

**Pull request merged**